File: src/main/java/project/rpg/effect/Damage.java
package project.rpg.effect;

import org.bukkit.entity.LivingEntity;

public class Damage extends EffectBase{

    @Override
    public void effect() {
        if (_entity.getHealth() > _amplifier) {
            _entity.damage(10.0);
        } else {
            _entity.damage(_entity.getHealth());
        }
    }

    public Damage(LivingEntity entity, int amplifier) {
        super(entity, 1, amplifier);
    }
}

File: src/main/kotlin/project/rpg/effects/objects/Damage.kt
package project.rpg.effects.objects

import org.bukkit.entity.LivingEntity
import project.rpg.effects.EffectBase

class Damage(entity: LivingEntity, amplifier: Int) : EffectBase(entity, 1, amplifier) {
    override fun effect(durationMillis: Long) {
        if (entity.health > amplifier) {
            entity.damage(10.0)
        } else {
            entity.damage(entity.health)
        }
    }
}
